site/content/posts/2023-08-09-v0.25.0-impersonation-proxy-with-external-certs.md

```diff
@@ -0,0 +1,337 @@
+cert-manager v1.12.3
+````
+
+Set up kind to run with Contour, using the example kind cluster configuration file provided by Contour.
+
+```shell
+$ wget https://raw.githubusercontent.com/projectcontour/contour/main/examples/kind/kind-expose-port.yaml
+$ kind create cluster \
+    --config kind-expose-port.yaml \
+    --name kind-with-contour \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+```
+
+Install Contour (see https://projectcontour.io/getting-started/ for more details).
+
+```shell
+# From https://projectcontour.io/getting-started/
+$ kubectl apply \
+    --filename https://projectcontour.io/quickstart/contour.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+# Verify that the Contour pods are ready
+$ kubectl get pods \
+    --namespace projectcontour \
+    --output wide \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+```
+
+Install Pinniped’s local-user-authenticator and add some sample users (see https://pinniped.dev/docs/tutorials/concierge-only-demo/ for more details).
+
+```shell
+# Install Pinniped's local-user-authenticator
+$ kubectl apply \
+    --filename https://get.pinniped.dev/v0.25.0/install-local-user-authenticator.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+# Create a local user "pinny" with password "password123" and group "group-for-mtls".
+# Each secret in this namespace acts like a user definition.
+$ kubectl create secret generic pinny \
+    --namespace local-user-authenticator \
+    --from-literal=groups=group-for-mtls \
+    --from-literal=passwordHash=$(htpasswd -nbBC 10 x password123 | sed -e "s/^x://") \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+# We'll need the CA bundle of the local-user-authenticator service to configure the Concierge's WebhookAuthenticator.
+# Just make sure this next command does print out the TLS secret, which can take a few seconds to generate.
+$ kubectl get secret local-user-authenticator-tls-serving-certificate \
+    --namespace local-user-authenticator \
+    --output jsonpath={.data.caCertificate} \
+    --kubeconfig kind-with-contour.kubeconfig.yaml \
+    | tee local-user-authenticator-ca.pem.b64
+```
+
+Install Pinniped’s Concierge:
+
+```shell
+$ kubectl apply \
+    --filename https://get.pinniped.dev/v0.25.0/install-pinniped-concierge-crds.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+
+$ kubectl apply \
+    --filename https://get.pinniped.dev/v0.25.0/install-pinniped-concierge-resources.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+```
+
+Install `cert-manager`:
+
+```shell
+$ kubectl apply \
+    --filename https://github.com/cert-manager/cert-manager/releases/download/v1.12.3/cert-manager.yaml \
+    --kubeconfig kind-with-contour.kubeconfig.yaml
+```
+
+Configure a `cert-manager` certificate for the impersonation proxy to serve TLS.
+
+Note that this section bootstraps a CA `Issuer` used to issue leaf certificates that can be used to serve TLS.
```

Comment:
Can we add something like:

> For this demonstration we will be using Cert Manager to simulate our own Public Key Infrastructure (PKI).  We will bootstraps a CA `Issuer`.....

Gets those critical keywords into our text for search magic. 🧙‍♂️

Reply:
Resolving locally...